alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDocLevelMonitorFanOutAction.kt

```diff
@@ -1107,6 +1113,41 @@ class TransportDocLevelMonitorFanOutAction
         return response.hits
     }
 
+    private suspend fun fetchDocForFinding(
+        index: String,
+        docId: String,
+        fields: List<String>
+    ): Map<String, List<Any>> {
+        val qb = QueryBuilders.matchQuery("_id", docId)
+
+        val request: SearchRequest = SearchRequest()
+            .indices(index)
+            .source(
+                SearchSourceBuilder()
+                    .version(true)
+                    .sort("_seq_no", SortOrder.ASC)
+                    .seqNoAndPrimaryTerm(true)
+                    .query(qb)
+                    .fetchSource(false)
+            )
+
+        if (fields.isNotEmpty()) {
```

Comment:
this check should be preliminary to assert that if its empty this search is useless